pkg/generators/openapi_test.go

```diff
@@ -343,6 +343,239 @@ Type: []string{"object"},
 `, funcBuffer.String())
 }
 
+func TestNestedStruct(t *testing.T) {
+	callErr, funcErr, assert, callBuffer, funcBuffer := testOpenAPITypeWriter(t, `
+package foo
+
+// Blah demonstrate a struct with struct field.
+type Nested struct {
+  // A simple string
+  String string
+}
+
+type Blah struct {
+  // A struct field
+  Field Nested
+}
+	`)
+	if callErr != nil {
+		t.Fatal(callErr)
+	}
+	if funcErr != nil {
+		t.Fatal(funcErr)
+	}
+	assert.Equal(`"base/foo.Blah": schema_base_foo_Blah(ref),
+`, callBuffer.String())
+	assert.Equal(`func schema_base_foo_Blah(ref common.ReferenceCallback) common.OpenAPIDefinition {
+return common.OpenAPIDefinition{
+Schema: spec.Schema{
+SchemaProps: spec.SchemaProps{
+Type: []string{"object"},
+Properties: map[string]spec.Schema{
+"Field": {
+SchemaProps: spec.SchemaProps{
+Description: "A struct field",
+Ref: ref("base/foo.Nested"),
+},
+},
+},
+Required: []string{"Field"},
+},
+},
+Dependencies: []string{
+"base/foo.Nested",},
+}
+}
+
+`, funcBuffer.String())
+}
+
+func TestNestedStructPointer(t *testing.T) {
+	callErr, funcErr, assert, callBuffer, funcBuffer := testOpenAPITypeWriter(t, `
+package foo
+
+// Blah demonstrate a struct with struct pointer field.
+type Nested struct {
+  // A simple string
+  String string
+}
+
+type Blah struct {
+  // A struct pointer field
+  Field *Nested
+}
+	`)
+	if callErr != nil {
+		t.Fatal(callErr)
+	}
+	if funcErr != nil {
+		t.Fatal(funcErr)
+	}
+	assert.Equal(`"base/foo.Blah": schema_base_foo_Blah(ref),
+`, callBuffer.String())
+	assert.Equal(`func schema_base_foo_Blah(ref common.ReferenceCallback) common.OpenAPIDefinition {
+return common.OpenAPIDefinition{
+Schema: spec.Schema{
+SchemaProps: spec.SchemaProps{
+Type: []string{"object"},
+Properties: map[string]spec.Schema{
+"Field": {
+SchemaProps: spec.SchemaProps{
+Description: "A struct pointer field",
+Ref: ref("base/foo.Nested"),
+},
+},
+},
+Required: []string{"Field"},
+},
+},
+Dependencies: []string{
+"base/foo.Nested",},
+}
+}
+
+`, funcBuffer.String())
+}
+
+func TestEmbeddedStruct(t *testing.T) {
+	callErr, funcErr, assert, callBuffer, funcBuffer := testOpenAPITypeWriter(t, `
+package foo
+
+// Blah demonstrate a struct with embedded struct field.
+type Nested struct {
+  // A simple string
+  String string
+}
+
+type Blah struct {
+  // An embedded struct field
+  Nested
+}
+	`)
+	if callErr != nil {
+		t.Fatal(callErr)
+	}
+	if funcErr != nil {
+		t.Fatal(funcErr)
+	}
+	assert.Equal(`"base/foo.Blah": schema_base_foo_Blah(ref),
+`, callBuffer.String())
+	assert.Equal(`func schema_base_foo_Blah(ref common.ReferenceCallback) common.OpenAPIDefinition {
+return common.OpenAPIDefinition{
+Schema: spec.Schema{
+SchemaProps: spec.SchemaProps{
+Type: []string{"object"},
+Properties: map[string]spec.Schema{
+"Nested": {
```

Comment:
is this correct? I don't think Nested should be the name of a field?

Reply:
Because it is not indicated as inlined the name of the embedded type should be the name of the field. That's consistent with the usage in the go language, where fields of embedded fields can be denoted by the the type name followed by the desired sub field

Reply:
That's still the old behaviour, I just added tests for all cases, including the already correctly handled ones.

Reply:
Oh yes, that's probably because it doesn't have `json:",inline"`.